ccm_web/client/src/components/CreateSectionWidget.tsx

```diff
@@ -53,18 +55,22 @@ function CreateSectionWidget (props: CreateSectionWidgetProps): JSX.Element {
           .then(newSections => {
             props.onSectionCreated(newSections[0])
             setNewSectionName('')
-          }).catch(() => {
-            enqueueSnackbar('Error adding section', {
-              variant: 'error'
-            })
+          }).catch((e: Error) => {
+            if (hasUnauthorized([e])) {
+              location.href = '/'
+            } else {
+              enqueueSnackbar('Error adding section', { variant: 'error' })
+            }
           })
       } else {
         errorAlert(errors)
       }
-    }).catch(() => {
-      enqueueSnackbar('Error validating section name', {
-        variant: 'error'
-      })
+    }).catch((e: Error) => {
+      if (hasUnauthorized([e])) {
+        location.href = '/'
+      } else {
+        enqueueSnackbar('Error validating section name', { variant: 'error' })
+      }
```

Comment:
My approach using `hasUnauthorized` becomes a bit clunkier when we're not using `usePromise`, because the errors are not all available in the same scope. I guess this works for now though? I didn't want to mess with what @chrisrrowland had put together too much.

Reply:
No longer relevant, since I changed the implementation.